feat(NavigationMenu): allow tooltip usage in `horizontal` orientation

EvanSchleret opened the pull request:
### 🔗 Linked issue

Resolves #5666 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [X] 👌 Enhancement (improving an existing functionality)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

With UNavigationMenu, it’s possible to display tooltips, but only in a horizontal orientation. In certain use cases, some users would like to show tooltips for items that don’t have a label for the navigation menu (as seen in the linked issue). This PR addresses this requirement.

### 📝 Checklist

- [X] I have linked an issue or discussion.
- [X] I have updated the documentation accordingly.

EvanSchleret committed:
fix(navigationmenu): make possible to use tooltip in horizontal orientation

EvanSchleret commented:
I just read the comment from the Vercel bot, and he’s absolutely right. If you’re open to accepting this change, Benjamin, would you prefer a different approach? If so, could you please provide me some information so I can propose a compromise that you’d find acceptable?

benjamincanac commented:
I think it is correct as well as I don't think we should simply enable tooltips in horizontal orientation, it will conflict with the actual menu. We should rather handle `collapsed` in horizontal orientation I guess.

EvanSchleret committed:
fix(navigationmenu): make possible to use tooltip in horizontal orientation

EvanSchleret commented:
Would this approach be more effective to you?

EvanSchleret commented:
Hey Benjamin, I hope you enjoyed the holiday season. I'm coming back to this PR to see if it suited you and if you could therefore possibly accept it. Happy new year !

benjamincanac committed:
Merge branch 'v4' into v4

benjamincanac reviewed (commented):
@EvanSchleret Happy new year as well! 😊

I'm not sure to understand your changes, the `hidden` class is already applied in `src/theme/navigation-menu.ts` when `collapsed` is true. I think we can keep the same behavior in `vertical` and `horizontal` orientations, there's no need for a tooltip if the label is displayed anyway.

EvanSchleret committed:
fix(navigation-menu): keep collapsed label visibility theme-driven

remove component-level horizontal label/trailing hiding logic and rely on navigation-menu theme classes for collapsed behavior

restore docs wording for collapsed + tooltip to match vertical orientation semantics

EvanSchleret committed:
fix: fix conflict

EvanSchleret commented:
Hey @benjamincanac, I hope you’re doing well.

I completely forgot about my PR and your last comment. I made some changes based on your feedback. Would this be a better solution for you?

Thanks for your answer!

EvanSchleret committed:
fix: remove hasToolTip const

EvanSchleret committed:
fix(navigation-menu): enable per-item tooltip collapse in horizontal mode

(Sorry, I committed the wrong conflict resolution previously ...)

this commit restores the intended fix:
- in horizontal + collapsed mode, only tooltip-enabled items collapse to icon-only
- non-tooltip items keep their label/trailing visible
- tooltip rendering in horizontal mode is enabled only for tooltip-enabled items

EvanSchleret committed:
merge: sync nuxtui/v4 into v4 and keep horizontal tooltip behavior

benjamincanac committed:
Merge branch 'v4' into v4

benjamincanac committed:
fix(NavigationMenu): allow tooltip usage in `horizontal` orientation

benjamincanac commented:
@EvanSchleret I simplified the approach so `tooltip` works in horizontal mode without needing `collapsed` (which is vertical-only). Let me know what you think!

EvanSchleret commented:
Your implementation is actually much better. I’d love to adopt this kind of approach when coding, but I tend to overengineer things, haha! Do you rely "onyl" on your experience, or do you have any advice? If so, I’d like to contribute again to something else here or somewhere else. Thanks!

benjamincanac committed:
playground(navigation-menu): add tooltip for example

benjamincanac committed:
fix(NavigationMenu): update tsdoc and delayDuration only on vertical

benjamincanac committed:
docs(navigation-menu): allow changing orientation in example

benjamincanac committed:
fix(NavigationMenu): only override tooltip props on vertical orientation

benjamincanac committed:
docs(navigation-menu): add notes

benjamincanac commented:
@EvanSchleret I think it mostly comes down to consistency. With 125+ components in Nuxt UI, the same patterns keep coming back so you naturally end up keeping things simple although the NavigationMenu component is very complicated 😅

I've added a `contributing` skill you can invoke inside the `nuxt/ui` repo that knows all the conventions, should help for next time. Would love to see more contributions from you, you can DM me on Discord if you need any guidance!

EvanSchleret commented:
Thank you! I’ll definitely do it if needed :)
